swap/src/cli/api/request.rs

```diff
@@ -471,6 +498,260 @@ impl Request for GetMoneroAddressesArgs {
     }
 }
 
+#[typeshare]
+#[derive(Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct GetMoneroHistoryArgs;
+
+#[typeshare]
+#[derive(Serialize, Clone, Deserialize, Debug)]
+pub struct GetMoneroHistoryResponse {
+    pub transactions: Vec<monero_sys::TransactionInfo>,
+}
+
+impl Request for GetMoneroHistoryArgs {
+    type Response = GetMoneroHistoryResponse;
+
+    async fn request(self, ctx: Arc<Context>) -> Result<Self::Response> {
+        let wallet = ctx
+            .monero_manager
+            .as_ref()
+            .context("Monero wallet manager not available")?;
+        let wallet = wallet.main_wallet().await;
+
+        let transactions = wallet.history().await;
+        Ok(GetMoneroHistoryResponse { transactions })
+    }
+}
+
+#[typeshare]
+#[derive(Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct GetMoneroMainAddressArgs;
+
+#[typeshare]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct GetMoneroMainAddressResponse {
+    #[typeshare(serialized_as = "String")]
+    pub address: monero::Address,
+}
+
+impl Request for GetMoneroMainAddressArgs {
+    type Response = GetMoneroMainAddressResponse;
+
+    async fn request(self, ctx: Arc<Context>) -> Result<Self::Response> {
+        let wallet = ctx
+            .monero_manager
+            .as_ref()
+            .context("Monero wallet manager not available")?;
+        let wallet = wallet.main_wallet().await;
+        let address = wallet.main_address().await;
+        Ok(GetMoneroMainAddressResponse { address })
+    }
+}
+
+#[typeshare]
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(tag = "type", content = "height")]
+pub enum SetRestoreHeightArgs {
+    #[typeshare(serialized_as = "number")]
+    Height(u32),
+    #[typeshare(serialized_as = "string")]
+    Date(String),
+}
+
+#[typeshare]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct SetRestoreHeightResponse {
+    pub success: bool,
+}
+
+impl Request for SetRestoreHeightArgs {
+    type Response = SetRestoreHeightResponse;
+
+    async fn request(self, ctx: Arc<Context>) -> Result<Self::Response> {
+        let wallet = ctx
+            .monero_manager
+            .as_ref()
+            .context("Monero wallet manager not available")?;
+        let wallet = wallet.main_wallet().await;
+
+        let height = match self {
+            SetRestoreHeightArgs::Height(height) => height as u64,
+            SetRestoreHeightArgs::Date(date) => {
```

Comment:
We shouldn't parse the datestring here. Do that in the frontend instead. Change the `SetRestoreHeightArgs::Date` to accept a `{year: u32, month: u32, day: u32`. Change the `monero/SetRestoreHeightModal.tsx` to pass in just the date in that format into the function in `rpc.ts`.

No need to serialize into a string here and then go back again.